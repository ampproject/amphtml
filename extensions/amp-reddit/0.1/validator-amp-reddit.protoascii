#
# Copyright 2016 The AMP HTML Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the license.
#

tags: {  # amp-reddit
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "SCRIPT"
  spec_name: "amp-reddit extension .js script"
  also_requires_tag: "amp-reddit"
  mandatory_parent: "HEAD"
  attrs: {
    name: "async"
    mandatory: true
    value: ""
  }
  attrs: {
    name: "custom-element"
    mandatory: true
    value: "amp-reddit"
    dispatch_key: true
  }
  attrs: {
    name: "src"
    mandatory: true
    value_regex: "https://cdn\\.ampproject\\.org/v0/amp-reddit-(latest|0\\.1).js"
  }
  attrs: {
    name: "type"
    value: "text/javascript"
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "."
      error_message: "contents"
    }
  }
  spec_url: "https://github.com/ampproject/amphtml/blob/master/extensions/amp-reddit/amp-reddit.md"
}
tags: {  # <amp-reddit>
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "AMP-REDDIT"
  disallowed_ancestor: "AMP-SIDEBAR"
  also_requires_tag: "amp-reddit extension .js script"
  attrs: {
    name: "data-embedlive"
    value_regex_casei: "(false|true)"
  }
  attrs: {
    name: "data-embedparent"
    value_regex_casei: "(false|true)"
  }
  attrs: {
    name: "data-embedtype"
    mandatory: true
    value_regex_casei: "(comment|post)"
  }
  attrs: {
    name: "data-src"
    mandatory: true
  }
  attr_lists: "extended-amp-global"
  spec_url: "https://github.com/ampproject/amphtml/blob/master/extensions/amp-reddit/amp-reddit.md"
  amp_layout: {
    supported_layouts: FILL
    supported_layouts: FIXED
    supported_layouts: FIXED_HEIGHT
    supported_layouts: FLEX_ITEM
    supported_layouts: NODISPLAY
    supported_layouts: RESPONSIVE
  }
}
