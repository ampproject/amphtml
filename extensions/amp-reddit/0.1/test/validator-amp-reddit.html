<!--
  Copyright 2018 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests for the amp-reddit tag. See the inline comments.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async custom-element="amp-reddit" src="https://cdn.ampproject.org/v0/amp-reddit-latest.js"></script>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>
  <!-- Valid: Embedding a reddit post -->
  <amp-reddit
    layout="responsive"
    width="300"
    height="400"
    data-embedtype="post"
    data-src="https://www.reddit.com/r/me_irl/comments/52rmir/me_irl/?ref=share&amp;ref_source=embed">
  </amp-reddit>
  <!-- Valid: Embedding a reddit comment -->
  <amp-reddit
    layout="responsive"
    width="400"
    height="400"
    data-embedtype="comment"
    data-src="https://www.reddit.com/r/sports/comments/54loj1/50_cents_awful_1st_pitch_given_a_historical/d8306kw"
    data-uuid="b1246282-bd7b-4778-8c5b-5b08ac0e175e"
    data-embedcreated="2016-09-26T21:26:17.823Z"
    data-embedparent="true"
    data-embedlive="true">
  </amp-reddit>

  <!-- Invalid: missing src attribute. -->
  <amp-reddit
    layout="responsive"
    width="300"
    height="400"
    data-embedtype="post">
  </amp-reddit>
</body>
</html>
