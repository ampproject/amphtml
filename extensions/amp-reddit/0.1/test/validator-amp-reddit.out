FAIL
|  <!--
|    Test Description:
|    Tests for the amp-reddit tag. See the inline comments.
|  -->
|  <!doctype html>
|  <html âš¡ lang="en">
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="./regular-html-version.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async custom-element="amp-reddit" src="https://cdn.ampproject.org/v0/amp-reddit-latest.js"></script>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|    <!-- Valid: Embedding a reddit post -->
|    <amp-reddit
|      layout="responsive"
|      width="300"
|      height="400"
|      data-embedtype="post"
|      data-src="https://www.reddit.com/r/me_irl/comments/52rmir/me_irl/?ref=share&amp;ref_source=embed">
|    </amp-reddit>
|    <!-- Valid: Embedding a reddit comment -->
|    <amp-reddit
|      layout="responsive"
|      width="400"
|      height="400"
|      data-embedtype="comment"
|      data-src="https://www.reddit.com/r/sports/comments/54loj1/50_cents_awful_1st_pitch_given_a_historical/d8306kw"
|      data-uuid="b1246282-bd7b-4778-8c5b-5b08ac0e175e"
|      data-embedcreated="2016-09-26T21:26:17.823Z"
|      data-embedparent="true"
|      data-embedlive="true">
|    </amp-reddit>
|
|    <!-- Invalid: missing src attribute. -->
|    <amp-reddit
>>   ^~~~~~~~~
amp-reddit/0.1/test/validator-amp-reddit.html:38:2 The mandatory attribute 'data-src' is missing in tag 'amp-reddit'. (see https://amp.dev/documentation/components/amp-reddit)
|      layout="responsive"
|      width="300"
|      height="400"
|      data-embedtype="post">
|    </amp-reddit>
|  </body>
|  </html>
